test/position-managers/IncreaseLiquidity.t.sol

```diff
@@ -0,0 +1,327 @@
+
+        // swap to create fees
+        uint256 swapAmount = 0.001e18;
+        swap(key, true, -int256(swapAmount), ZERO_BYTES);
+        swap(key, false, -int256(swapAmount), ZERO_BYTES); // move the price back
+
+        // alice uses her exact fees to increase liquidity
+        // Slight error in this calculation vs. actual fees.. TODO: Fix this.
+        BalanceDelta feesOwedAlice = IPositionManager(lpm).getFeesOwed(manager, tokenIdAlice);
+        // Note: You can alternatively calculate Alice's fees owed from the swap amount, fee on the pool, and total liquidity in that range.
+        // swapAmount.mulWadDown(FEE_WAD).mulDivDown(liquidityAlice, liquidityAlice + liquidityBob);
+
+        (uint160 sqrtPriceX96,,,) = StateLibrary.getSlot0(manager, range.poolKey.toId());
+        uint256 liquidityDelta = LiquidityAmounts.getLiquidityForAmounts(
+            sqrtPriceX96,
+            TickMath.getSqrtPriceAtTick(range.tickLower),
+            TickMath.getSqrtPriceAtTick(range.tickUpper),
+            uint256(int256(feesOwedAlice.amount0())),
+            uint256(int256(feesOwedAlice.amount1()))
+        );
+
+        uint256 balance0BeforeAlice = currency0.balanceOf(alice);
+        uint256 balance1BeforeAlice = currency1.balanceOf(alice);
+
+        // TODO: Can we make this easier to re-invest fees, so that you don't need to know the exact collect amount?
+        Planner.Plan memory planner = Planner.init();
+        planner = planner.add(Actions.INCREASE, abi.encode(tokenIdAlice, liquidityDelta, ZERO_BYTES));
+        bytes memory calls = planner.finalize(range.poolKey);
+        vm.startPrank(alice);
+        lpm.modifyLiquidities(calls, _deadline);
+        vm.stopPrank();
+
+        // alice barely spent any tokens
+        // TODO: This is a case for not caring about dust left in pool manager :/
+        assertApproxEqAbs(balance0BeforeAlice, currency0.balanceOf(alice), tolerance);
+        assertApproxEqAbs(balance1BeforeAlice, currency1.balanceOf(alice), tolerance);
+    }
+
+    // uses donate to simulate fee revenue
+    function test_increaseLiquidity_withExactFees_donate() public {
+        // Alice and Bob provide liquidity on the range
+        // Alice uses her exact fees to increase liquidity (compounding)
+
+        uint256 liquidityAlice = 3_000e18;
+        uint256 liquidityBob = 1_000e18;
+
+        // alice provides liquidity
+        vm.prank(alice);
+        mint(range, liquidityAlice, alice, ZERO_BYTES);
+        uint256 tokenIdAlice = lpm.nextTokenId() - 1;
+
+        // bob provides liquidity
+        vm.prank(bob);
+        mint(range, liquidityBob, bob, ZERO_BYTES);
+
+        // donate to create fees
+        uint256 amountDonate = 0.2e18;
+        donateRouter.donate(key, 0.2e18, 0.2e18, ZERO_BYTES);
+
+        // subtract 1 cause we'd rather take than pay
+        uint256 feesAmount = amountDonate.mulDivDown(liquidityAlice, liquidityAlice + liquidityBob) - 1;
```

Comment:
doesnt the `div down` already handle the downward rounding for take vs settle? is it not enough?

Reply:
it didn't in this case. if i dont subtract 1 it does a pay/settle